languages/js/src/Query.ts

```diff
@@ -359,17 +363,44 @@ export class Query {
             }
             const { message } = event.data as Debug;
             if (message) console.log(message);
-            createInterface({
-              input: process.stdin,
-              output: process.stdout,
-              prompt: 'debug> ',
-              tabSize: 4,
-            }).on('line', (line: string) => {
-              const trimmed = line.trim().replace(/;+$/, '');
-              const command = this.#host.toPolar(trimmed);
-              this.#ffiQuery.debugCommand(JSON.stringify(command));
-              this.processMessages();
+
+            if (typeof rl === 'undefined') {
+              // @ts-ignore
+              rl = global.oso_repl_rl;
+              if (rl) {
+                const exitListeners = rl.listeners('exit') as (() => void)[];
+                rl.removeAllListeners('exit');
+
+                const lineListeners = rl.listeners('line') as (() => void)[];
+                rl.removeAllListeners('line');
+
+                const oldPrompt = rl.getPrompt();
+                rl.setPrompt('debug> ');
+
+                rl.prependOnceListener('exit', () => {
+                  rl!.setPrompt(oldPrompt);
+                  exitListeners.forEach((l) => rl!.addListener('exit', l));
+                  lineListeners.forEach((l) => rl!.addListener('line', l));
+                });
+              } else {
+                rl = createInterface({
+                  input: process.stdin,
+                  output: process.stdout,
+                  prompt: 'debug> ',
+                  tabSize: 4,
+                });
+              }
+            }
+
+            rl!.prompt();
+            const command: string = await new Promise(resolve => {
+              rl!.once('line', (line: string) => {
+                const trimmed = line.trim().replace(/;+$/, '');
+                resolve(trimmed)
+              });
             });
+            this.#ffiQuery.debugCommand(command);
```

Comment:
Could you emit the `exit` event on `rl` here instead of in the `Result` branch of the switch statement?